### PR TITLE
masked loyalty card number

### PR DESCRIPTION
**Pull request by sowen123**
Masked card number for `LoyaltyCardPayment`
Updated description of pin in `PaymentMethod` so customers know we are not expecting it in responses and will therefore not translate it in responses

### CODE REVIEW

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 This page describes changes to the Service Interface Base API implemented across different releases of the interface.
+## 3.34.0
+Released xx July 2021
```
Comment (PatrickColborne) [inline src/docs/devguide/hugo/site/content/release-notes.md:3]: Don't forget to update the date before releasing.

File: src/main/java/io/electrum/vas/model/PaymentMethod.java [hunk 2]
```diff
@@ -128,7 +128,8 @@ public PaymentMethod issuer(Institution issuer) {
       return this;
    }
 
-   @ApiModelProperty(value = "The PIN associated with this payment method. Various PIN formats are supported (clear, encrypted, hashed etc.)")
+   @ApiModelProperty(value = "The PIN associated with this payment method. Various PIN formats are supported (clear, "
```
Comment (PatrickColborne) [inline src/main/java/io/electrum/vas/model/PaymentMethod.java:131]: You've formatted a file that hasn't been edited. Perhaps revert these changes.

File: src/test/java/io/electrum/vas/model/NewModelTests.java [hunk 3]
```diff
@@ -176,7 +185,8 @@ public void testRecursiveValidationOnSubFields(Object objectWithInvalidSubField,
               {walletPayment, JsonUtil.readFileAsString(PayloadFileLocations.WALLET_PAYMENT, false)},
               {cardPayment, JsonUtil.readFileAsString(PayloadFileLocations.CARD_PAYMENT, false)},
               {qrPayment, JsonUtil.readFileAsString(PayloadFileLocations.QR_PAYMENT, false)},
-              {address, JsonUtil.readFileAsString(PayloadFileLocations.ADDRESS, false)}
+              {address, JsonUtil.readFileAsString(PayloadFileLocations.ADDRESS, false)},
+              {loyaltyCardPayment, JsonUtil.readFileAsString(PayloadFileLocations.LOYALTY_CARD_PAYMENT, false)},
```
Comment (PatrickColborne) [inline src/test/java/io/electrum/vas/model/NewModelTests.java:189]:
```suggestion
              {loyaltyCardPayment, JsonUtil.readFileAsString(PayloadFileLocations.LOYALTY_CARD_PAYMENT, false)}
```

File: src/test/java/io/electrum/vas/model/TestPanMasking.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            objectMapper.writeValueAsString(objectMapper.readValue(unmaskedValue, classOfObject)),
+            testString);
+   }
+
+   @DataProvider
+   public Object[][] maskedSerialisedObjectDataProvider() throws Exception {
+      return new Object[][] {
+            //@formatter:off
+            { loyaltyCardPayment,
+            JsonUtil.readFileAsString(PayloadFileLocations.LOYALTY_CARD_PAYMENT_MASKED, false) }
```
Comment (PatrickColborne) [inline src/test/java/io/electrum/vas/model/TestPanMasking.java:67]: I know that you may have copied this but you should probably put the `JsonUtil.readFileAsString()` method in the actual test rather than in the data provider. It would reduce repeated code and make it easier to read. Not really necessary to change for this I guess. Your choice.

Comment (PatrickColborne) [reply src/test/java/io/electrum/vas/model/TestPanMasking.java:67]: Obvs this would apply to other similar things.

**Review by PatrickColborne — APPROVED** [review]
Yeah that looks good. I have left some comments that aren't crucial so approved anyway.

File: src/main/java/io/electrum/vas/model/PaymentMethod.java (hunk 2 above)
Comment (sowen123) [reply src/main/java/io/electrum/vas/model/PaymentMethod.java:131]: Oh lol we needed to update the description of the PIN as well sorry I forgot to say that in the PR

File: src/test/java/io/electrum/vas/model/TestPanMasking.java (hunk 4 above)
Comment (sowen123) [reply src/test/java/io/electrum/vas/model/TestPanMasking.java:67]: Hmmm so the reason I don't want to do this is we are then forcing it to be in a file the whole time whereas now you can provide any string. Would it be alright if I leave this as is?

File: src/main/java/io/electrum/vas/model/PaymentMethod.java (hunk 2 above)
Comment (PatrickColborne) [reply src/main/java/io/electrum/vas/model/PaymentMethod.java:131]: Oh I see 🤦🏻‍♂️

File: src/test/java/io/electrum/vas/model/TestPanMasking.java (hunk 4 above)
Comment (PatrickColborne) [reply src/test/java/io/electrum/vas/model/TestPanMasking.java:67]: Yeah sure! It's just test stuff so not the end of the world.

**Review by PatrickColborne — APPROVED** [review]
I'm happy
